src/GovUk.Education.ExploreEducationStatistics.Content.Model/Repository/ReleaseVersionRepository.cs

```diff
@@ -172,6 +172,23 @@ public async Task<List<ReleaseVersion>> ListLatestReleaseVersions(CancellationTo
                releaseVersion.Version;
     }
 
+    public async Task<ReleaseVersion?> GetLatestReleaseVersionForParent(
+        Guid releaseId,
+        bool publishedOnly = false,
+        CancellationToken cancellationToken = default)
+    {
+        var maxVersion = await GetMaxVersionNumber(
+            releaseId,
+            publishedOnly: publishedOnly,
+            cancellationToken: cancellationToken);
+
+        return await _contentDbContext.ReleaseVersions
+            .Where(rv =>
+                rv.ReleaseId == releaseId
+                && rv.Version == maxVersion)
+            .FirstOrDefaultAsync(cancellationToken);
+    }
+
```

Comment:
You could consider a version of this which doesn't involve two queries, because I think this could all be done with the following join in a single query (I didn't try this out, but should be something along these lines!).

```
            var maxVersionsQueryable = _contentDbContext.ReleaseVersions
                .Where(releaseVersion => releaseVersion.ReleaseId == releaseId)
                .Where(releaseVersion => !publishedOnly || releaseVersion.Published.HasValue)
                .GroupBy(releaseVersion => releaseVersion.ReleaseId)
                .Select(groupedVersions => groupedVersions.Max(releaseVersion => releaseVersion.Version));

            return await _contentDbContext.ReleaseVersions
                .Where(releaseVersion => releaseVersion.ReleaseId == releaseId)
                .Join(maxVersionsQueryable,
                    releaseVersion => releaseVersion.Version,
                    maxVersion => maxVersion,
                    (releaseVersion, _) => releaseVersion)
                .SingleAsync(cancellationToken: cancellationToken);
```

Rather than make this change in `ReleaseVersionRepository` you could also consider adding it to `ReleaseVersionPredicate` in a method something like this:

```
    public static IQueryable<ReleaseVersion> LatestReleaseVersion(this IQueryable<ReleaseVersion> releaseVersions,
        Guid releaseId,
        bool publishedOnly = false)
    {
        var maxVersionsQueryable = releaseVersions
            .Where(releaseVersion => releaseVersion.ReleaseId == releaseId)
            .Where(releaseVersion => !publishedOnly || releaseVersion.Published.HasValue)
            .GroupBy(releaseVersion => releaseVersion.ReleaseId)
            .Select(groupedVersions => groupedVersions.Max(releaseVersion => releaseVersion.Version));

        return releaseVersions
            .Where(releaseVersion => releaseVersion.ReleaseId == releaseId)
            .Join(maxVersionsQueryable,
                releaseVersion => releaseVersion.Version,
                maxVersion => maxVersion,
                (releaseVersion, _) => releaseVersion);
    }
```

The repository method wouldn't even be needed as anything using it can just make a call like ` _contentDbContext.ReleaseVersions.LatestReleaseVersion(releaseId, publishedOnly: true)` without needing to even depend on the repository.

Reply:
I've gone with a nested query in the hopes of making it more readable - happy to switch to the join if you think that's preferable.

I've also made a change to `ListLatestReleaseVersions` to use `ReleaseVersionPredicates.LatestReleaseVersions` so we don't need to inject `IReleaseVersionRepository` into `PublicationService`.

Reply:
> I've also made a change to ListLatestReleaseVersions to use ReleaseVersionPredicates.LatestReleaseVersions so we don't need to inject IReleaseVersionRepository into PublicationService.

I think this is ok but is there much of an advantage to doing this?

`ReleaseVersionPredicates.LatestReleaseVersion` returns an unordered IQueryable so you've had to duplicate the logic for the reverse chronological ordering here.

I recently made an effort to tidy up as many of the places as possible where we were doing the ordering by year and then by time period to be contained within `ReleaseVersionRepository` methods. Except for methods in that repository which id's, they are using the method `OrderByReverseChronologicalOrder` to order results consistently. That logic is defined in a single place.

I think there's a couple of instances (MappingProfiles and MethodologyService) which were not easy to change. I'd have to revisit those to explain why.

Maybe you could reuse that logic here, or swap it back to how it was?

I feel like keeping the ordering logic in as fewer places as possible is a good idea, especially as we may swap it out to order by the release series at some point soon.

Reply:
I've switched back to the original